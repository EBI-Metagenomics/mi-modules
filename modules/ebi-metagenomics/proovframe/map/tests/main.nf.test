nextflow_process {

    name "Test Process PROOVFRAME_MAP"
    script "../main.nf"
    process "PROOVFRAME_MAP"

    tag "modules"
    tag "modules_nfcore"
    tag "proovframe"
    tag "proovframe/map"

    test("test - faa_fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/emales.faa", checkIfExists: true),
                    file("${moduleDir}/tests/fixtures/emales-reads-sim.fa", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.tsv[0][1]).text.contains('EMALE01_Cflag_c017B	EMALE01_Cflag_c017B_0005	73.5	646	154	12	1803	3727	7	642	1.02e-244	749	21173') },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }


    test("test - faa_fasta - stub") {
        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/emales.faa", checkIfExists: true),
                    file("${moduleDir}/tests/fixtures/emales-reads-sim.fa", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
