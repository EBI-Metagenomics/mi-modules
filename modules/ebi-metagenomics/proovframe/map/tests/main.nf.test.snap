{
    "test - faa_fasta": {
        "content": [
            [
                "versions.yml:md5,f49f29b8134fdb539364ee46855eb313"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T17:23:11.865268"
    },
    "test - faa_fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f49f29b8134fdb539364ee46855eb313"
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f49f29b8134fdb539364ee46855eb313"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T17:23:14.372527"
    }
}